### DISCUSSION TITLE
Get information about the index of a method in the containing type

### DISCUSSION THREAD

**Discussion by daveMueller** [dotnet/roslyn#70291] [Q&A]
Hi,

I wonder if it is possible to get the index of a method in the containing type after compilation? Can this information be found somewhere inside the PDB?
When I look into the documentation of the file
https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/CodeGen/DebugId.cs
it says something about 'the ordinal is included in the custom debug information record attached to the method'. But for instance when I try to analyze the assembly with `Mono.Cecil`, I can't find this information anywhere.

This is a bit related to another question I asked here a while back #68664.

**Comment by tmat** [comment]
> I wonder if it is possible to get the index of a method in the containing type after compilation? 

What purpose do you need this information for?

**Reply by daveMueller** [reply]
For instance if you do instrumentation on the assembly it would be more easy to map the generated synthesized method to the function that is using it. The synthesized method's name is a combination of the function name and mentioned ordinal, e.g. `<TestLambda>b__1`.
Now imagine having the following two functions:
```csharp
    public int TestLambda(string input)
    {
        System.Func<string, int> lambdaFunc = s => s.Length;
        return lambdaFunc(input);
    }

    public int TestLambda(string input, int value)
    {
        System.Func<string, int> lambdaFunc = s => s.Length;
        return lambdaFunc(input) + value;
    }
```
In the IL this would generate two types `<TestLambda>b__0` and `<TestLambda>b__1`. And now it is not that easy to determine which type was generated for which function. If the lambda is nested inside an async method it is even harder.
